tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+since we are using a VM anyway, let us take a look at how this done!
+
+We've already seen how to handle missing or conflicting dependecies, and we
+have also seen how to verify checksums -- these are both important aspects to
+installing packages from source, so let's put our new knowledge to work!
+
+#### Downloading the files
+
+Now that we're running an old version of `apache2` on our system, let us build
+the most up-to-date development version (`2.4.62` at the time of writing) from
+source. We are fortunate, since Apache2 has instructions for building the
+package on the
+[Apache2 download site](https://httpd.apache.org/download.cgi#apache24). The
+instructions will be different for every package (and not every package comes
+with instructions), but the principles are the same.
+
+All the package files are contained in a compressed single file called a
+"tarball" (which has the `.tar.gz` file extension). The maintainers should also
+provide the files needed to verify the package -- in the case of Apache2, they
+do! Let's first download the tarball:
+
+```bash
+wget https://dlcdn.apache.org/httpd/httpd-2.4.62.tar.gz
+```
+
+#### Verifying the checksums
+
+For this particular version of `apache` there is no MD5 checksum file so we
+can't use the `md5sum` tool we used earlier. But, there *are* SHA256 and SHA512
+checksum files. Let's download the SHA256 checksum file -- it's advised to
+download these from the official source, so let's do that:
+
+```bash
+wget https://downloads.apache.org/httpd/httpd-2.4.62.tar.gz.sha256
+```
+
+Then we can verify the package download against that checksum using the
+`sha256sum` tool:
+
+```bash
+sha256sum -c httpd-2.4.62.tar.gz.sha256
+```
+
+Which should return:
+
+```bash
+httpd-2.4.62.tar.gz: OK
+```
+
+This tells us that the package file is validated. If the output returned
+`FAILED` then we would know something is wrong with it. In that case, we would
+want to download the file again (from a trusted source, preferably). We
+definitely wouldn't want to install it if the check fails! 
+
+In our case, since we obtained the package from a trusted source (the official
+website) and the checksums matched, we can proceed with the installation.
+
+#### Unpacking and installing
+
+First,  let's "un-tar" the tarball:
+
+```bash
+tar -xvzf httpd-2.4.62.tar.gz
+```
+
+This will unpack the contents of the tarball into a directory of the same name,
+which we can change to:
+
+```bash
+cd httpd-2.4.62
+```
+
+If we run `ls --all` we can see a list of all the files that are contained in
+the package.
+
+There are a couple of dependencies that we'll want to install that don't come
+packaged with the development version, so let's install those now:
+
+```bash
+sudo apt install autoconf libtool-bin
+```
+
+Next we'll want to configure the source tree -- the prefix is the most
+important option, so let's set that (for the sake of this example, we'll just
+set it to the default location):
```

Comment:
why do we want to configure the source tree? What is the source tree and why does it need to be configured? (You lightly explain below, but I think context should be provided before they run the command)